keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt

```diff
@@ -99,13 +97,14 @@ class ArtifactMetadataService(
         .retryOnException { e: Throwable? -> e is HttpException }
         .build()
     )
-    return Try.ofSupplier(Retry.decorateSupplier(retry
-    ) {
-      runBlocking {
-        buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
-      }
-    }).get()
+
+    //decorate the suspended function and execute it
+    return retry.decorateSuspendFunction {
+      buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
+    } ()
```

Comment:
I think this variant is a little more readable:
```suggestion
    return retry.executeSuspendFunction {
      buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
    }
```

Reply:
good call!